File: benchmarks/head_to_head/python/daftar_operasi.py
"""Workload "daftar_operasi" -- versi Python. N SAMA dengan versi Isoteri
(n=20000), tapi build list pakai append() -- idiomatik Python, O(1)
amortized. Beda karakteristik build-list ini SENGAJA dibiarkan & dicatat
jujur di README.md, bukan dipaksa sama supaya "adil" secara artifisial."""


def konversi_ke_rupiah(kg):
    return kg * 5000


def di_atas_ambang(x):
    return x > 1000000


if __name__ == "__main__":
    n = 20000
    data = []
    for i in range(n):
        data.append(i % 500)

    nilai_rupiah = list(map(konversi_ke_rupiah, data))
    signifikan = list(filter(di_atas_ambang, nilai_rupiah))
    print(sum(signifikan))

File: benchmarks/head_to_head/node/daftar_operasi.js
// Workload "daftar_operasi" -- versi Node.js. N SAMA dengan versi Isoteri
// (n=20000), tapi build array pakai push() -- idiomatik JS, O(1) amortized.
// Beda karakteristik build-list ini SENGAJA dibiarkan & dicatat jujur di
// README.md, bukan dipaksa sama supaya "adil" secara artifisial.

function konversiKeRupiah(kg) {
  return kg * 5000;
}

function diAtasAmbang(x) {
  return x > 1000000;
}

const n = 20000;
const data = [];
for (let i = 0; i < n; i++) {
  data.push(i % 500);
}

const nilaiRupiah = data.map(konversiKeRupiah);
const signifikan = nilaiRupiah.filter(diAtasAmbang);
const total = signifikan.reduce((a, b) => a + b, 0);
console.log(total);
